### ISSUE TITLE
Remote pane: what should be shown?

### ISSUE THREAD

**Issue by justinclift** [sqlitebrowser/sqlitebrowser#2297]
The Remote pane currently (v3.12.0) just shows two things:

* A list of all users with public databases (on DBHub.io), ordered from the newest change to the oldest
* The list of all databases the user has uploaded to DBHub.io

With the the list of users with public databases now getting pretty long, it's hard to use the Remote pane effectively. :frowning:

Instead, we should probably update the Remote tab to show:

* A curated list  <-- probably a good default.  eg something like:
  * Learning SQL
    * [CS50x](https://cs50.harvard.edu/x/2020/psets/7/movies/) - the database ([movies.zip](https://cdn.cs50.net/2019/fall/psets/7/movies/movies.zip)) used in this
    * Ditto for (eg): https://www.codecademy.com/learn/paths/analyze-data-with-sql
    * Ditto: https://www.sqlitetutorial.net/sqlite-sample-database/
    * Ditto: https://datarockie.teachable.com/p/intro-sql-for-data-analysis <-- non-english though
    * Ditto: https://www.datacamp.com/community/tutorials/sqlite-in-python
    * and others.
  * Interesting data sets (TBD)

* People's own databases.

* All public user uploaded databases - like the current approach, but with a filter like the top of the DB4S Browse Data tab, so people can more easily find users/databases they're after.
  * Ideally we'd have some kind of "search" interface, but that's likely a pita SysAdmin wise for the backend (eg needing an Elasticsearch cluster), so I'd skip it at least for now.

For the curated list, it'd probably require us to say Hi to each of the course authors about getting their database onto DBHub, and so they can include the info about it in their download instructions (if desired).  It might also mean they get a useful boost in people learning about their course too, if we have each course entry in DB4S have some kind of a clickable link that opens a web browser window to the course overview page.

Thoughts?

**Comment by justinclift** [comment]
@sayakpaul You're the author of the DataCamp SQLite tutorial in the list above.  Is there a database that students doing the course should be using, and do you reckon DataCamp would be ok with it being included? :smile:

**Comment by sayakpaul** [comment]
@justinclift there's no database as such. As long as they are following the tutorial with an SQLite engine, they should be good to go. 

> do you reckon DataCamp would be ok with it being included?

As long as it is duly cited, I don't think there would be any problem. But for more explicit permission, I would recommend sending an email to community@datacamp.com or to ryan@datacamp.com.

**Comment by justinclift** [comment]
Cool.  Thanks heaps @sayakpaul. :smile:

**Comment by justinclift** [comment]
@dmalan Same thought as above.  Do you reckon this would be useful for the CS50x course?  I sent you an email about it, but it's probably easier to communicate here. :smile:

**Comment by justinclift** [comment]
Interestingly, Udemy seems to have a course in Spanish which might be suitable too:

&nbsp; &nbsp; https://www.udemy.com/course/sql-curso-completo-de-sql-aprende-desde-cero/

No idea what the course content is like though. :wink:

**Comment by chrisjlocke** [comment]
[Suggestion] If we had a public list, then what about some sort of 'voting' against it, so dead ones can be voted down, and good ones can be voted up, ala Reddit?  Nothing worse than having a public database only for it to (maybe) get overwritten at some point or have invalid data, while a more 'interesting one' just sits at the bottom of the list.  You could even have 'this weeks most popular database' 😉 
Having some 'manual' override would be good, for example, if someone uploaded a Covid-data related database, you'd maybe want that (temporarily) at the top of the list, but you wouldn't want it at the top next year, etc...

**Comment by justinclift** [comment]
Sure, we can get fancy.  But lets get the basics figured out first? :smile:

**Comment by justinclift** [comment]
Discussing with @dmalan via email yesterday, the movies.zip used in the CS50x course comes from IMDb.  [IMDb's licensing](https://help.imdb.com/article/imdb/general-information/can-i-use-imdb-data-in-my-software/G5JTRESSHJBBHTGX#) is extremely restrictive, so that database is unusable for us:

> The data can only be used for **personal *and* non-commercial** use and must not be altered/republished/resold/repurposed to create any kind of online/offline database of movie information (except for **individual personal** use).

**Comment by MKleusberg** [comment]
I have just added an alternative view to the Remote dock which shows all the databases you have downloaded or uploaded. It's still not perfect but already makes the Remote dock much more usable.

**Comment by justinclift** [comment]
**Awesome!!**

**Comment by justinclift** [comment]
Tried it out just now.  The new tab which shows downloaded databases looks like it'll be useful. :smile:

With this tabbed approach, could you also add a tab called something like "Yours", which shows just the databases in the remote users folder?  So they don't need to find their own folder in the long list of all public ones.

The new Current Database tab looks like it'll be useful for displaying properties too, so good idea. :smile:

Probably need some minor tweaking to make sense when no database is loaded through:

![2020-07-11-Remote_tab-635dba93c295147e57cc927f81d205e95b723154-no_database_loaded](https://user-images.githubusercontent.com/406299/87213429-3d426d00-c368-11ea-9bbe-8c4da9c998a3.png)

When seeing that for the first time, I was thinking the UI was broken.  When a database is loaded though, it makes sense:

![2020-07-11-Remote_tab-635dba93c295147e57cc927f81d205e95b723154-example_database_loaded](https://user-images.githubusercontent.com/406299/87213464-8abeda00-c368-11ea-8db3-075ea3e52b8a.png)

Maybe turn it into an obvious table, or have placeholder text like "No database loaded" be shown instead?

**Comment by chrisjlocke** [comment]
> could you also add a tab called something like "Yours"

Could we do the Microsoft approach of 'My Databases' instead of 'Yours', otherwise people will be looking for 'Yours' then 'Mine' 😉 
Actually, you're right.  Just checked the 'Preferences' and the 'Remote' tab, and we have 'Your certificates', so we do do 'your' and not 'my'.  I'll shut up then! 😉 

> or have placeholder text like "No database loaded" 

Again, was looking around looking for other examples of this, and all the textboxes in 'Preferences' have a border (but they all mainly expect text input) or are greyed out.  Maybe we could have textboxes rather than labels, and agree about the 'No database loaded' placeholder text.

**Comment by dmalan** [comment]
> @dmalan Same thought as above. Do you reckon this would be useful for the CS50x course? I sent you an email about it, but it's probably easier to communicate here. 😄

Hi @justinclift, just to close the loop here too! For our course purposes, I don't think the sample databases need to be built into the UI, to be honest. In fact, I think we'd prefer that students realize that, in general, step 1 is to download (or create) the database, and step 2 is to open it in an app like DB Browser, fwiw!

**Comment by justinclift** [comment]
@dmalan No worries. :smile:

In theory (!) one of the hopefully-useful future capabilities of the remote server will be collaboration capabilties (eg showing diffs between databases, merging diffs in GitHub-like-PR-fashion, full traceability of all changes).  But we're not there yet, even though the foundations are mostly in place now.

When those collaboration pieces are usefully functional, we're hoping there's more reason for people to usem.  Until then, it's just effectively remote storage with version control. :wink:

**Comment by justinclift** [comment]
> Could we do the Microsoft approach of 'My Databases' ...

Makes sense.  Personally, I'm not bothered by what it's called, as long as the concept of "this is your stuff on the remote server" is clear for people. :smile:

**Comment by chrisjlocke** [comment]
>  I'm not bothered by what it's called

Well, I vote to call it 'George' then.  😄

**Comment by justinclift** [comment]
Sure, but you're going to confuse a lot of people. :wink:

**Comment by MKleusberg** [comment]
> With this tabbed approach, could you also add a tab called something like "Yours", which shows just the databases in the remote users folder? So they don't need to find their own folder in the long list of all public ones.

That's no problem. But instead of adding another button we could make sure the user's directory is always on the top of the list. That would be a server-side thing though. What do you think?

> The new Current Database tab looks like it'll be useful for displaying properties too, so good idea. smile
> Probably need some minor tweaking to make sense when no database is loaded through:

Good point. Before changing it to a table view let's think about what information should be shown. I was thinking about a commit tree at some point and definitely some buttons. Anybody wants to make a mockup? :smile:

**Comment by justinclift** [comment]
> ... we could make sure the user's directory is always on the top of the list.

Yeah, we can give that a go.  It'll probably be good enough. :smile:

I won't get it done today, as I'm kind of burnt out atm.  Likely next weekend though, unless you want to update the server code yourself? :smile:

Actual deployment of updated server code is still manual at the moment, though 99.9% automated for anything that doesn't involve database schema changes.

**Comment by MKleusberg** [comment]
I'll see if I have time for that this week :smile: 

By the way, I have added a list of all branches and commits to the 'Current Database' section of the Remote dock. It doesn't do anything useful yet but, again, it's a start :wink:

**Comment by justinclift** [comment]
> ... we could make sure the user's directory is always on the top of the list.

Just implemented that, and pushed it to production. eg it's live now.

Hopefully that makes things easier to work with for people. :smile:

**Comment by MKleusberg** [comment]
Awesome :smile: DB4S also automatically expands the current user's directory. Hopefully that helps too.
